nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/versioning/events/CompoundDocumentVersioningListener.java

```diff
@@ -120,15 +120,19 @@ protected void handleCheckedInEvent() {
         } else {
             return;
         }
-        Optional<VersionModel> latestVersion = coreSession.
-                getVersionsForDocument(document.getRef()).stream().max(Comparator.comparing(VersionModel::getCreated));
+        Optional<VersionModel> latestVersion = coreSession.getVersionsForDocument(document.getRef())
+                                                          .stream()
+                                                          .max(Comparator.comparing(VersionModel::getCreated));
```

Comment:
`latestVersion` is always used with `get()` which is not recommended, and we can all it directly here instead of later. Could you add a `orElseThrow()` call here please?

Reply:
I created https://hyland.atlassian.net/browse/NXP-33050 -> This addon needs further clean up IMO

I started a branch https://github.com/nuxeo/nuxeo-compound-documents/compare/task-NXP-32776-align2025...fix-NXP-33050-versions-2025?expand=1 which I'd like to finalize in a second round of clean up.

This work would probably benefit `2023` and `2021`